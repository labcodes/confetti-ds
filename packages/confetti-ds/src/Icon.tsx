import React from "react";

import { IconTypes, IconColorTypes } from "./constants";
import {
  ArrowLeft20Regular,
  ArrowRight20Regular,
  ArrowUp20Regular,
  ArrowDown20Regular,
  ArrowSort20Regular,
  Star20Regular,
  BookOpen20Regular,
  CloudArrowUp20Regular,
  Key20Regular,
  DocumentHeart20Regular,
  TextFont20Regular,
  ChatEmpty20Regular,
  CalligraphyPen20Regular,
  CaretLeft20Regular,
  CaretRight20Regular,
  CaretUp20Regular,
  CaretDown20Regular,
  ChevronUp20Regular,
  ChevronDown20Regular,
  ChevronRight20Regular,
  ChevronLeft20Regular,
  ArrowClockwise20Regular,
  Heart20Regular,
  Javascript20Regular,
  Clock20Regular,
  Dismiss20Regular,
  Add20Regular,
  Subtract20Regular,
  Checkmark20Regular,
  ArrowMinimize20Regular,
  ArrowMaximize20Regular,
  Backspace20Regular,
  Important20Regular,
  Edit20Regular,
  ArrowTrending20Regular,
  Lightbulb20Regular,
  ProjectionScreen20Regular,
  Eye20Regular,
  EyeOff20Regular,
  Search20Regular,
  Delete20Regular,
  Navigation20Regular,
  Link20Regular,
  Filter20Regular,
  Settings20Regular,
  Bookmark20Regular,
  WindowApps20Regular,
  Cube20Regular,
  CodeBlock20Regular,
  ErrorCircle20Regular,
  SignOut20Regular,
  StackStar20Regular,
  CalendarLtr20Regular,
  Money20Regular,
  Open20Regular,
  GridDots20Regular,
  Phone20Regular,
  Wallet20Regular,
  CubeRotate20Regular,
  Cloud20Regular,
  Mic20Regular,
  CloudFlow20Regular,
} from "@fluentui/react-icons";
export interface IconProps {
  /** Type of the Icon to be rendered. */
  type: IconTypes;
  /** Add a class name to make custom changes */
  className?: string;
}
export default function Icon({ type, className }: IconProps) {
  let IconComponent;

  if (type === "arrow-left") IconComponent = ArrowLeft20Regular;

  if (type === "arrow-right") IconComponent = ArrowRight20Regular;

  if (type === "arrow-up") IconComponent = ArrowUp20Regular;

  if (type === "arrow-down") IconComponent = ArrowDown20Regular;

  if (type === "sort") IconComponent = ArrowSort20Regular;

  if (type === "star") IconComponent = Star20Regular;

  if (type === "book-open") IconComponent = BookOpen20Regular;

  if (type === "upload") IconComponent = CloudArrowUp20Regular;

  if (type === "key") IconComponent = Key20Regular;

  if (type === "sheet") IconComponent = DocumentHeart20Regular;

  if (type === "type") IconComponent = TextFont20Regular;

  if (type === "message") IconComponent = ChatEmpty20Regular;

  if (type === "pen") IconComponent = CalligraphyPen20Regular;

  if (type === "select-right") IconComponent = CaretRight20Regular;

  if (type === "select-left") IconComponent = CaretLeft20Regular;

  if (type === "select-up") IconComponent = CaretUp20Regular;

  if (type === "select-down") IconComponent = CaretDown20Regular;

  if (type === "chevron-up") IconComponent = ChevronUp20Regular;

  if (type === "chevron-down") IconComponent = ChevronDown20Regular;

  if (type === "chevron-left") IconComponent = ChevronLeft20Regular;

  if (type === "chevron-right") IconComponent = ChevronRight20Regular;

  if (type === "reload") IconComponent = ArrowClockwise20Regular;

  if (type === "heart") IconComponent = Heart20Regular;

  if (type === "js") IconComponent = Javascript20Regular;

  if (type === "code") IconComponent = CodeBlock20Regular;

  if (type === "clock") IconComponent = Clock20Regular;

  if (type === "dismiss") IconComponent = Dismiss20Regular;

  if (type === "plus") IconComponent = Add20Regular;

  if (type === "minus") IconComponent = Subtract20Regular;

  if (type === "confirm") IconComponent = Checkmark20Regular;

  if (type === "menu-collapse") IconComponent = ArrowMinimize20Regular;

  if (type === "menu-expand") IconComponent = ArrowMaximize20Regular;

  if (type === "delete") IconComponent = Backspace20Regular;

  if (type === "exclamation") IconComponent = Important20Regular;

  if (type === "edit") IconComponent = Edit20Regular;

  if (type === "growth") IconComponent = ArrowTrending20Regular;

  if (type === "lamp") IconComponent = Lightbulb20Regular;

  if (type === "presentation") IconComponent = ProjectionScreen20Regular;

  if (type === "add-circle") IconComponent = Add20Regular;

  if (type === "eye-opened") IconComponent = Eye20Regular;

  if (type === "eye-closed") IconComponent = EyeOff20Regular;

  if (type === "lupe") IconComponent = Search20Regular;

  if (type === "trash") IconComponent = Delete20Regular;

  if (type === "menu-default") IconComponent = Navigation20Regular;

  if (type === "link") IconComponent = Link20Regular;

  if (type === "filter") IconComponent = Filter20Regular;

  if (type === "gear") IconComponent = Settings20Regular;

  if (type === "bookmark") IconComponent = Bookmark20Regular;

  if (type === "web-app") IconComponent = WindowApps20Regular;

  if (type === "pack") IconComponent = Cube20Regular;

  if (type === "chevrons") IconComponent = CodeBlock20Regular;

  if (type === "star") IconComponent = ErrorCircle20Regular;

  if (type === "logout") IconComponent = SignOut20Regular;

  if (type === "track") IconComponent = StackStar20Regular;

  if (type === "calendar") IconComponent = CalendarLtr20Regular;

  if (type === "coin") IconComponent = Money20Regular;

  if (type === "external") IconComponent = Open20Regular;

  if (type === "workspace") IconComponent = GridDots20Regular;

  if (type === "phone") IconComponent = Phone20Regular;

  if (type === "wallet") IconComponent = Wallet20Regular;

  if (type === "continuous-delivery") IconComponent = CubeRotate20Regular;

  if (type === "cloud") IconComponent = Cloud20Regular;

  if (type === "microphone") IconComponent = Mic20Regular;

  if (type === "api") IconComponent = CloudFlow20Regular;

  if (!IconComponent) return null;

  return (
    <span>
      <IconComponent
        className={
          `lab-icon lab-icon--${type}` + `${className ? ` ${className}` : ""}`
        }
      />
    </span>
  );
}
