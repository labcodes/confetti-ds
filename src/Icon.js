import React from "react";
import PropTypes from "prop-types";

import { ICON_TYPES } from "./constants";

export default class Icon extends React.Component {
  static propTypes = {
    /** Type of the Icon to be rendered. */
    type: PropTypes.oneOf(ICON_TYPES).isRequired,
    /** Color of the rendered Icon. */
    color: PropTypes.oneOf([
      "white",
      "black-75",
      "mineral-10",
      "mineral-20",
      "mineral-30",
      "mineral-40",
      "mineral-60",
      "mineral-70",
      "mineral-80",
      "mineral-90",
      "teal-40",
      "teal-60",
      "teal-70",
      "purple-40",
      "purple-60",
      "purple-70",
    ]),
    /** Size of the rendered Icon, petit = 16px, small = 20px. If omitted will render at 24px. */
    size: PropTypes.oneOf(["small", "petit"]),
    /** Add a class name to make custom changes */
    className: PropTypes.string,
  };

  static defaultProps = {
    color: undefined,
    size: undefined,
    className: undefined,
  };

  render() {
    const { type, color, size, className } = this.props;
    return (
      <span
        className={
          `lab-icon lab-icon--${type}` +
          `${color ? ` lab-icon--${color}` : ""}` +
          `${size ? ` lab-icon--${size}` : ""}` +
          `${className ? ` ${className}` : ""}`
        }
      />
    );
  }
}
