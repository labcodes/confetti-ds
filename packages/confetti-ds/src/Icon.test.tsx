import React from "react";
import renderer from "react-test-renderer";

import Icon from "./Icon";

describe("Icon", () => {
  it("renders with base props", () => {
    const renderedComponent = renderer.create(<Icon type="Cloud" />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
