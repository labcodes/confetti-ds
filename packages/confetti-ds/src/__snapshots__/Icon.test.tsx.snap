// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders with base props 1`] = `
<span>
  <svg
    aria-hidden={true}
    className="lab-icon lab-icon--cloud ___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
    fill="currentColor"
    height={20}
    viewBox="0 0 20 20"
    width={20}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 4c2.82 0 4.41 1.92 4.65 4.25h.07A3.33 3.33 0 0 1 18 11.62 3.33 3.33 0 0 1 14.72 15H5.28A3.33 3.33 0 0 1 2 11.62c0-1.8 1.37-3.27 3.1-3.37h.25C5.6 5.9 7.18 4 10 4Zm0 1C7.89 5 6.55 6.32 6.35 8.35a1 1 0 0 1-1 .9h-.07A2.33 2.33 0 0 0 3 11.62 2.33 2.33 0 0 0 5.28 14h9.44A2.33 2.33 0 0 0 17 11.62a2.33 2.33 0 0 0-2.28-2.37h-.07a1 1 0 0 1-1-.9C13.45 6.32 12.11 5 10 5Z"
      fill="currentColor"
    />
  </svg>
</span>
`;
